### PR TITLE
Fixing template property deprecation, so the addon can work with latest ember v2.10.0

### PR DESCRIPTION
**Pull request by waleedq**
Hello, 

It appears that the component's template property is deprecated from v1.x api but there was no any deprecation messages about using it, and it have been replace with the layout property. to fix this issue in v2.10.0 the partial helper should be used to render the given template inside the dialog layout

this PR should fix this issue https://github.com/wheely/ember-dialog/issues/87

Thank You,

### CODE REVIEW

**Comment by ivandokov** [comment]
Maybe you should also change the README file to show the layout changes

**Comment by ajile** [comment]
Recently I do not have spare time to review and apply PR. I will try to do that tomorrow. I'm sorry about it.

**Comment by ajile** [comment]
I tried to run some examples from the dummy (in russian version) and found a problem - `partial` helper is not working with precompiled template. Occurs an [internal exception](https://github.com/tildeio/glimmer/blob/master/packages/glimmer-runtime/lib/compiled/opcodes/partial.ts#L25):

```
Error while processing route: index Could not find a partial named "[object Object]"
Error: Could not find a partial named "[object Object]"
```

That happens because the `partial` helper expects a string, but gets an object (precompiled template).  You can reproduce it so:

```es6
import Ember from "ember";
import hbs from 'htmlbars-inline-precompile';

export default Ember.Controller.extend({
  init() {
    this.get("dialog").alert(hbs`Hello world`);
    return this._super(...arguments);
  }
});
```

I need time to research Glimmer to find way to fix it. For now I don't know how Glimmer works.

**Comment by ajile** [comment]
I've reproduced problem on [ember-twiddle](https://ember-twiddle.com/113a3e5468f8ef4d989016d70d13a54b?openFiles=components.my-component.js).

**Comment by ajile** [comment]
After talking with community in slack, I've found solution to paste prerendered template into the presenter's body. I've created one more component (`presenter-body`) that gets template as layout. From now if template is a string - it's introduced into the presenter's body as a `partial`, otherwise it introduced as a `presenter-body` component.

@waleedq, could you review and accept PR waleedq/ember-dialog#6 if all fine.

File: tests/unit/services/dialog-test.js [hunk 1]
```diff
@@ -7,9 +7,6 @@ moduleFor('service:dialog', 'Unit | Service | dialog', {
   needs: ['component:presenter'],
   setup() {
     this.service = this.subject();
-  },
```
Comment (ajile) [inline tests/unit/services/dialog-test.js]: It's removed by mistake, i suppose?

Comment (ivandokov) [inline tests/unit/services/dialog-test.js]: This is how git displays the change. There is one `}` below. The code is fine.

Comment (ajile) [inline tests/unit/services/dialog-test.js]: I meant that the `teardown` method was removed in this commit. I pulled `commit` into the local machine. There is really no `teardown`.

**Comment by waleedq** [comment]
@ajile Sorry for being late, i've tested the PR and it works totally fine it's merged now, please check it out and let me know

**Comment by ajile** [comment]
@waleedq thank you! Everything works well. I'm going to merge PR and release stable version.

**Pull request merged**